packages/proto-signing/src/registry.ts

```diff
@@ -8,24 +9,111 @@ export interface GeneratedType {
   readonly decode: (reader: protobuf.Reader | Uint8Array, length?: number) => any;
 }
 
+export type EncodeObject = {
+  readonly typeUrl: string;
+  readonly value: any;
+};
+
+export type DecodeObject = {
+  readonly typeUrl: string;
+  readonly value: Uint8Array;
+};
+
+export type TxBodyValue = {
+  readonly messages: readonly EncodeObject[];
+  readonly memo?: string;
+  readonly timeoutHeight?: number;
+  readonly extensionOptions?: readonly any[];
+  readonly nonCriticalExtensionOptions?: readonly any[];
+};
+
+const defaultTypeUrls = {
+  cosmosCoin: "/cosmos.Coin",
+  cosmosMsgSend: "/cosmos.bank.MsgSend",
+  cosmosTxBody: "/cosmos.tx.TxBody",
+  googleAny: "/google.protobuf.Any",
+};
+
 export class Registry {
   private readonly types: Map<string, GeneratedType>;
 
   constructor(customTypes: Iterable<[string, GeneratedType]> = []) {
+    const { cosmosCoin, cosmosMsgSend, cosmosTxBody, googleAny } = defaultTypeUrls;
     this.types = new Map<string, GeneratedType>([
-      ["/cosmos.Coin", cosmosSdk.v1.Coin],
-      ["/cosmos.bank.MsgSend", cosmosSdk.x.bank.v1.MsgSend],
-      ["/cosmos.tx.TxBody", cosmosSdk.tx.v1.TxBody],
-      ["/google.protobuf.Any", google.protobuf.Any],
+      [cosmosCoin, cosmosSdk.v1.Coin],
+      [cosmosMsgSend, cosmosSdk.x.bank.v1.MsgSend],
+      [cosmosTxBody, cosmosSdk.tx.v1.TxBody],
+      [googleAny, google.protobuf.Any],
```

Comment:
I think we should remove the `TxBody` and `Any` here, as we don't need to encode it as an `Any`. `TxBody` lives at a different level and is not as dynamic. The only thing we really need the registry for is message types and types used by messages (such as Coin).